pkg/storage/replica.go

```diff
@@ -5148,6 +5148,51 @@ func (r *Replica) sendRaftMessage(ctx context.Context, msg raftpb.Message) {
 		return
 	}
 
+	if msg.Type == raftpb.MsgAppResp && msg.Reject {
+		r.mu.RLock()
+		desc := r.Desc()
+		ticks := r.mu.ticks
+		r.mu.RUnlock()
+
+		if msg.RejectHint == 0 && ticks < r.store.cfg.RaftPostSplitSuppressSnapshotTicks {
```

Comment:
Don’t you want to check at least one of those aforementioned conditions besides ticks? Because ticks resets on node reboot, no?

Reply:
I’m on mobile so I can’t actually check this right now.

Reply:
In checking that `RejectHint == 0` the replica basically tells us that it has empty state, so it implies `!IsInitialized()`. Or am I misunderstanding you?
I think that's what @bdarnell meant when he said

> Just one of the first three checks should be sufficient (in combination with ticks < 10).

Reply:
Oh, I see, I totally misunderstood what RejectHint was. I thought it was some internal Raft thing; didn't realize it was the last Raft index.

I do think `RejectHint == 0` is the least obvious condition to check (`!r.IsInitialized() && r.mu.ticks < 10` is just clearer to me), but I'm also relatively unfamiliar with Raft so if this reads better to you, leave it!